### ISSUE TITLE
[Bug]: waiter is paused when running sysbench oltp_read_only test and tps nearly 0

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#14150]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch Name

1.1-dev

### Commit ID

9d120d4

### Other Environment Information

```Markdown
- Hardware parameters:
stanlone 64 256G
- OS type:
- Others:
```

### Actual Behavior

job link:
https://github.com/matrixorigin/mo-nightly-regression/actions/runs/7491252658/job/20392150035

<img width="1211" alt="image" src="https://github.com/matrixorigin/matrixone/assets/47846308/38d827df-a526-44e1-9bc7-4178734cb936">

mo-log:
[sysbench1000w.tar.gz](https://github.com/matrixorigin/matrixone/files/13910826/sysbench1000w.tar.gz)

### Steps to Reproduce

```Markdown
1. run a mo on stanlone mode
2. run sysbench oltp oltp_read_only case
```

**Comment by heni02** [comment]
#14254

**Comment by aunjgr** [comment]
PR merged

**Comment by aressu1985** [comment]
testing

**Comment by aressu1985** [comment]
fixed: 83b8499b3727ba8fb2814c9b9900145014c7e11f

[sudong@mo-srv-128 mo-sysbench]$ sysbench  --mysql-host=127.0.0.1 --mysql-port=6001 --mysql-user=dump --mysql-password=111 --db-ps-mode=disable oltp_read_only.lua --mysql-db=sysbench_1000w_auto_index --tables=10 --table_size=10000000  --time=60 --report-interval=2 --point_selects=1 --threads=100 run
sysbench 1.0.20 (using system LuaJIT 2.1.0-beta3)

Running the test with following options:
Number of threads: 100
Report intermediate results every 2 second(s)
Initializing random number generator from current time


Initializing worker threads...

Threads started!

[ 2s ] thds: 100 tps: 1922.98 qps: 13625.71 (r/w/o: 9729.79/0.00/3895.91) lat (ms,95%): 77.19 err/s: 0.00 reconn/s: 0.00
[ 4s ] thds: 100 tps: 2052.53 qps: 14351.68 (r/w/o: 10247.13/0.00/4104.55) lat (ms,95%): 69.29 err/s: 0.00 reconn/s: 0.00
[ 6s ] thds: 100 tps: 2058.31 qps: 14410.20 (r/w/o: 10293.07/0.00/4117.13) lat (ms,95%): 68.05 err/s: 0.00 reconn/s: 0.00
[ 8s ] thds: 100 tps: 2074.98 qps: 14525.89 (r/w/o: 10375.92/0.00/4149.97) lat (ms,95%): 68.05 err/s: 0.00 reconn/s: 0.00
[ 10s ] thds: 100 tps: 2061.44 qps: 14436.10 (r/w/o: 10313.22/0.00/4122.89) lat (ms,95%): 68.05 err/s: 0.00 reconn/s: 0.00
[ 12s ] thds: 100 tps: 2061.97 qps: 14433.80 (r/w/o: 10311.86/0.00/4121.94) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 14s ] thds: 100 tps: 2063.52 qps: 14441.63 (r/w/o: 10313.10/0.00/4128.54) lat (ms,95%): 68.05 err/s: 0.00 reconn/s: 0.00
[ 16s ] thds: 100 tps: 2059.85 qps: 14429.44 (r/w/o: 10310.24/0.00/4119.20) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 18s ] thds: 100 tps: 2054.23 qps: 14370.12 (r/w/o: 10260.66/0.00/4109.46) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 20s ] thds: 100 tps: 2087.29 qps: 14600.50 (r/w/o: 10425.93/0.00/4174.58) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 22s ] thds: 100 tps: 2074.99 qps: 14533.43 (r/w/o: 10383.45/0.00/4149.98) lat (ms,95%): 65.65 err/s: 0.00 reconn/s: 0.00
[ 24s ] thds: 100 tps: 2081.55 qps: 14584.35 (r/w/o: 10421.25/0.00/4163.10) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 26s ] thds: 100 tps: 2041.69 qps: 14285.36 (r/w/o: 10201.97/0.00/4083.39) lat (ms,95%): 68.05 err/s: 0.00 reconn/s: 0.00
[ 28s ] thds: 100 tps: 2072.43 qps: 14494.50 (r/w/o: 10349.65/0.00/4144.86) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 30s ] thds: 100 tps: 2057.62 qps: 14402.84 (r/w/o: 10287.60/0.00/4115.24) lat (ms,95%): 68.05 err/s: 0.00 reconn/s: 0.00
[ 32s ] thds: 100 tps: 2061.97 qps: 14443.76 (r/w/o: 10321.33/0.00/4122.43) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 34s ] thds: 100 tps: 2104.38 qps: 14707.66 (r/w/o: 10497.40/0.00/4210.26) lat (ms,95%): 65.65 err/s: 0.00 reconn/s: 0.00
[ 36s ] thds: 100 tps: 2061.52 qps: 14453.66 (r/w/o: 10331.11/0.00/4122.54) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 38s ] thds: 100 tps: 2077.16 qps: 14516.63 (r/w/o: 10362.30/0.00/4154.32) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 40s ] thds: 100 tps: 2039.51 qps: 14296.07 (r/w/o: 10216.55/0.00/4079.52) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 42s ] thds: 100 tps: 2079.77 qps: 14549.86 (r/w/o: 10390.33/0.00/4159.53) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 44s ] thds: 100 tps: 2071.01 qps: 14514.10 (r/w/o: 10372.07/0.00/4142.03) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 46s ] thds: 100 tps: 2073.92 qps: 14517.45 (r/w/o: 10369.60/0.00/4147.84) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 48s ] thds: 100 tps: 2065.95 qps: 14450.66 (r/w/o: 10318.76/0.00/4131.90) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 50s ] thds: 100 tps: 2062.84 qps: 14450.37 (r/w/o: 10324.69/0.00/4125.68) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 52s ] thds: 100 tps: 2075.89 qps: 14530.25 (r/w/o: 10378.46/0.00/4151.78) lat (ms,95%): 65.65 err/s: 0.00 reconn/s: 0.00
[ 54s ] thds: 100 tps: 2092.84 qps: 14657.38 (r/w/o: 10471.70/0.00/4185.68) lat (ms,95%): 65.65 err/s: 0.00 reconn/s: 0.00
[ 56s ] thds: 100 tps: 2069.24 qps: 14473.67 (r/w/o: 10335.70/0.00/4137.98) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 58s ] thds: 100 tps: 2067.40 qps: 14479.27 (r/w/o: 10343.98/0.00/4135.29) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
[ 60s ] thds: 100 tps: 2064.23 qps: 14437.12 (r/w/o: 10308.16/0.00/4128.96) lat (ms,95%): 66.84 err/s: 0.00 reconn/s: 0.00
SQL statistics:
    queries performed:
        read:                            619450
        write:                           0
        other:                           247780
        total:                           867230
    transactions:                        123890 (2063.76 per sec.)
    queries:                             867230 (14446.34 per sec.)
    ignored errors:                      0      (0.00 per sec.)
    reconnects:                          0      (0.00 per sec.)

General statistics:
    total time:                          60.0294s
    total number of events:              123890

Latency (ms):
         min:                                   17.20
         avg:                                   48.44
         max:                                  123.78
         95th percentile:                       66.84
         sum:                              6001093.76

Threads fairness:
    events (avg/stddev):           1238.9000/4.55
    execution time (avg/stddev):   60.0109/0.01
